airflow/models/taskinstance.py

```diff
@@ -352,7 +353,26 @@ def _run_raw_task(
         if not test_mode:
             _add_log(event=ti.state, task_instance=ti, session=session)
             if ti.state == TaskInstanceState.SUCCESS:
-                ti._register_asset_changes(events=context["outlet_events"], session=session)
+                task_outlets = []
+                outlet_events = []
+                events = context["outlet_events"]
+                for obj in ti.task.outlets or []:
+                    # Lineage can have other types of objects besides assets
+                    asset_type = type(obj).__name__
+                    if isinstance(obj, Asset):
+                        task_outlets.append(AssetProfile(name=obj.name, uri=obj.uri, asset_type=asset_type))
+                        outlet_events.append(attrs.asdict(events[obj]))  # type: ignore
+                    elif isinstance(obj, AssetNameRef):
+                        task_outlets.append(AssetProfile(name=obj.name, asset_type=asset_type))
+                        outlet_events.append(attrs.asdict(events))  # type: ignore
+                    elif isinstance(obj, AssetUriRef):
+                        task_outlets.append(AssetProfile(uri=obj.uri, asset_type=asset_type))
+                        outlet_events.append(attrs.asdict(events))  # type: ignore
+                    elif isinstance(obj, AssetAlias):
+                        task_outlets.append(AssetProfile(asset_type=asset_type))
+                        for asset_alias_event in events[obj].asset_alias_events:
+                            outlet_events.append(attrs.asdict(asset_alias_event))
+                TaskInstance._register_asset_changes_int(ti, task_outlets, outlet_events, session=session)
```

Comment:
This makes nuking https://github.com/apache/airflow/pull/45924/files#diff-62f7d8a52fefdb8e05d4f040c6d3459b4a56fe46976c24f68843dbaeb5a98487L2736-L2742 possible. Not a big fan that we have to iterate here to calculate types and populate:
```
                task_outlets = []
                outlet_events = []
```

And iterate again here: 
https://github.com/apache/airflow/pull/45924/files#diff-62f7d8a52fefdb8e05d4f040c6d3459b4a56fe46976c24f68843dbaeb5a98487R2781-R2795

But should be ok for now